llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -54759,6 +54923,10 @@ static SDValue combineX86AddSub(SDNode *N, SelectionDAG &DAG,
   bool IsSub = X86ISD::SUB == N->getOpcode();
   unsigned GenericOpc = IsSub ? ISD::SUB : ISD::ADD;
 
+  if (IsSub && isOneConstant(N->getOperand(1)) && !N->hasAnyUseOfValue(0))
+    if (SDValue CMP = combineX86SubCmpForFlags(N, SDValue(N, 1), DAG, DCI, ST))
+      return CMP;
```

Comment:
This should use DCI.CombineTo accepting two SDValues, the first one of which should be the first (unused) result of SUB. The SUB will be removed later on.

Reply:
What does this `DCI.CombineTo` look like? I am trying replace the second result of `SUB` (old flag) with the result of CCMP (new flag). Not understand why you suggest the first parameter is the first (unused) result of SUB.

Reply:
SUB has two results, CMP has one result, which corresponds to the second result of the SUB.
Something like `return DCI.CombineTo(N, SDValue(N, 0), CMP)` would make the removed assert not trigger.
But I see you've found another way of silencing it.